File: POO/Constructores_Clases/Class/JAVA/Producto.java
public class Producto {
    private int id;
    private String nombre;
    private double precio;

    // Constructor con parámetros
    public Producto(int id, String nombre, double precio) {
        this.id = id;
        this.nombre = nombre;
        this.precio = precio;
    }

    // Constructor de copia
    public Producto(Producto other) {
        this.id = other.id;
        this.nombre = other.nombre;
        this.precio = other.precio;
    }

    public void mostrar() {
        System.out.printf("ID: %d, Producto: %s, Precio: $%.2f%n", 
            id, nombre, precio);
    }

    public static void main(String[] args) {
        Producto original = new Producto(1, "Laptop", 1500.00);
        Producto copia = new Producto(original);

        original.mostrar();
        copia.mostrar();
    }
}

File: POO/Constructores_Clases/Class/GO/Producto.go
package main

import "fmt"

type Producto struct {
    id     int
    nombre string
    precio float64
}

// Constructor con parámetros
func NewProducto(id int, nombre string, precio float64) *Producto {
    return &Producto{
        id:     id,
        nombre: nombre,
        precio: precio,
    }
}

// Constructor de copia
func NewProductoFrom(other *Producto) *Producto {
    return &Producto{
        id:     other.id,
        nombre: other.nombre,
        precio: other.precio,
    }
}

func (p *Producto) Mostrar() {
    fmt.Printf("ID: %d, Producto: %s, Precio: $%.2f\n", 
        p.id, p.nombre, p.precio)
}

func main() {
    original := NewProducto(1, "Laptop", 1500.00)
    copia := NewProductoFrom(original)
    
    original.Mostrar()
    copia.Mostrar()
}